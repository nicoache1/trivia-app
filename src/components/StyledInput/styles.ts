import { StyleSheet } from 'react-native'

export const styles = StyleSheet.create({
  input: {
    backgroundColor: 'transparent',
  },
  labelContainer: {
    paddingHorizontal: 10,
    paddingVertical: 5,
  },
})
