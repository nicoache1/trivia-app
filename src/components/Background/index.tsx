import React from 'react'
import { useWindowDimensions, View } from 'react-native'
import Svg, { Path } from 'react-native-svg'

import { styles } from './styles'

export const Background: React.FC<{}> = ({}) => {
  const { width } = useWindowDimensions()

  return (
    <View style={styles.vector}>
      <Svg
        width={width}
        height="812"
        viewBox="0 0 375 812"
        fill="none"
        preserveAspectRatio="xMinYMin slice">
        <Path
          d="M-757.5 612.5L-1210 243.5V948.5H459V264.5L112 673L-335.5 -76L-757.5 612.5Z"
          fill="#142664"
          fillOpacity="0.3"
        />
      </Svg>
    </View>
  )
}
