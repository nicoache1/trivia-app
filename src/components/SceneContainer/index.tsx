import * as React from 'react'
import { StatusBar, StatusBarProps, StyleProp, ViewStyle } from 'react-native'
import { Edge, SafeAreaView } from 'react-native-safe-area-context'
import { useTheme } from 'src/styles/Theme'

import { styles } from './styles'

interface LayoutProps {
  edges?: ReadonlyArray<Edge>
  style?: StyleProp<ViewStyle>
  children: React.ReactNode
  barStyle?: StatusBarProps['barStyle']
}

export const SceneContainer: React.FunctionComponent<LayoutProps> = ({
  edges = ['bottom'],
  style,
  children,
  barStyle = 'default',
}) => {
  const { Theme } = useTheme()

  return (
    <SafeAreaView
      style={[styles.container, { backgroundColor: Theme.colors.WHITE }, style]}
      edges={edges}>
      <StatusBar barStyle={barStyle} animated={true} />
      {children}
    </SafeAreaView>
  )
}
