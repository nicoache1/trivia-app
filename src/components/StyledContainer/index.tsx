import { concat } from 'lodash'
import React from 'react'
import { StyleProp, View, ViewProps, ViewStyle } from 'react-native'

import { useTheme } from '../../styles/Theme'
import { ColorPalette } from '../../styles/types'

type CustomViewProps = {
  color?: ColorPalette
  children?: React.ReactNode
} & ViewProps

export const StyledContainer: React.FC<CustomViewProps> = props => {
  const { Theme: ActualTheme } = useTheme()

  const { color, style, children } = props
  const themedColor = color
    ? ActualTheme.colors[color]
    : ActualTheme.colors.TRANSPARENT

  const contextStyle: StyleProp<ViewStyle> = {
    backgroundColor: themedColor,
    paddingHorizontal: ActualTheme.spacing.HORIZONTAL_SCREEN_PADDING,
    paddingVertical: 10,
  }

  const mergedStyle: StyleProp<ViewStyle> = style
    ? style instanceof Array
      ? concat([contextStyle], style)
      : { ...contextStyle, ...(style as any) }
    : contextStyle

  return (
    <View {...props} style={mergedStyle}>
      {children}
    </View>
  )
}
