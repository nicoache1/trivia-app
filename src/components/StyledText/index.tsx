import { concat } from 'lodash'
import React from 'react'
import { StyleProp, Text, TextProps, TextStyle } from 'react-native'

import { useTheme } from '../../styles/Theme'
import { AppTypography, ColorPalette } from '../../styles/types'

type TextCustomProps = {
  typography?: AppTypography
  color?: ColorPalette
  children: React.ReactNode
} & TextProps

export const StyledText: React.FC<TextCustomProps> = props => {
  const { Theme: ActualTheme } = useTheme()
  const {
    color = ColorPalette.BLACK,
    children,
    typography = AppTypography.P1_REGULAR,
    style,
  } = props

  const themedColor = color
    ? ActualTheme.colors[color]
    : ActualTheme.colors.BLACK

  const themedTypography = ActualTheme.typography[typography]

  const contextStyle: StyleProp<TextStyle> = {
    color: themedColor,
    ...themedTypography,
  }

  const mergedStyle: StyleProp<TextStyle> = style
    ? style instanceof Array
      ? concat([contextStyle], style)
      : { ...contextStyle, ...(style as any) }
    : contextStyle

  return (
    <Text {...props} style={mergedStyle} accessible={true}>
      {children}
    </Text>
  )
}
