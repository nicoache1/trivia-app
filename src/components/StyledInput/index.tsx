import React from 'react'
import { TextInputProps } from 'react-native'
import { HelperText, TextInput } from 'react-native-paper'

import { useTheme } from '../../styles/Theme'
import { StyledContainer } from '../StyledContainer'
import { styles } from './styles'

interface StyledInputProps extends TextInputProps {
  helperText?: string
  label?: string
  error?: boolean
}

export const StyledInput: React.FC<StyledInputProps> = ({
  placeholder,
  label,
  error = false,
  helperText,
  ...props
}) => {
  const { Theme } = useTheme()

  const borderColor = error ? Theme.colors.ERROR : Theme.colors.WHITE

  return (
    <StyledContainer>
      <TextInput
        {...props}
        label={label}
        style={[
          styles.input,
          {
            borderColor,
          },
        ]}
        activeOutlineColor={Theme.colors.WHITE}
        placeholderTextColor={Theme.colors.WHITE}
        placeholder={placeholder}
        selectionColor={Theme.colors.WHITE}
        theme={{
          colors: {
            accent: borderColor,
            disabled: borderColor,
            placeholder: borderColor,
            primary: borderColor,
            text: borderColor,
          },
        }}
        mode={'flat'}
      />
      {!!helperText && (
        <HelperText type={'error'} visible={error}>
          {helperText}
        </HelperText>
      )}
    </StyledContainer>
  )
}
