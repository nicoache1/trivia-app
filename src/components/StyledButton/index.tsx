import React from 'react'
import { PressableProps, StyleProp, ViewStyle } from 'react-native'
import { Pressable } from 'react-native'
import Animated from 'react-native-reanimated'

import { useTheme } from '../../styles/Theme'
import { AppTypography } from '../../styles/types'
import { StyledText } from '../StyledText'
import { ButtonMode } from './ButtonMode'
import { buttonStyle, styles, textStyle } from './styles'

interface Props extends PressableProps {
  label: string
  mode?: ButtonMode
  customStyles?: StyleProp<ViewStyle>
}

export const StyledButton: React.FC<Props> = ({
  label,
  mode = ButtonMode.LIGHT,
  onPress,
  customStyles,
}) => {
  const { Theme } = useTheme()
  return (
    <Pressable
      style={({ pressed }) => [styles.button, { opacity: pressed ? 0.5 : 1.0 }]}
      onPress={onPress}>
      <Animated.View
        style={[
          styles.button,
          customStyles !== undefined ? customStyles : buttonStyle(Theme)[mode],
        ]}>
        <StyledText
          style={[styles.buttonText, textStyle(Theme)[mode]]}
          typography={AppTypography.P1_BOLD}>
          {label}
        </StyledText>
      </Animated.View>
    </Pressable>
  )
}
