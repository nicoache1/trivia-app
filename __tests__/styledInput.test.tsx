import { render } from '@testing-library/react-native'
import React from 'react'
import { StyledInput } from 'src/components/StyledInput'

describe('Test StyledInput', () => {
  test('Expect StyledInput to be rendered correctly with some label', async () => {
    const renderedSnapshot = render(<StyledInput label="Label" />).toJSON()

    expect(renderedSnapshot).toMatchSnapshot()
  })

  test('Expect StyledInput to be rendered with error text', async () => {
    const renderedSnapshot = render(
      <StyledInput error={true} helperText={'Some error'} />,
    ).toJSON()

    expect(renderedSnapshot).toMatchSnapshot()
  })

  test('Expect StyledInput to not be rendered with error text', async () => {
    const renderedSnapshot = render(
      <StyledInput error={false} helperText={'Some error'} />,
    ).toJSON()

    expect(renderedSnapshot).toMatchSnapshot()
  })
})
