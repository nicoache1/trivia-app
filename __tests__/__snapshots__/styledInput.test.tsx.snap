// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test StyledInput Expect StyledInput to be rendered correctly with some label 1`] = `
<View
  style={
    Object {
      "backgroundColor": "transparent",
      "paddingHorizontal": 20,
      "paddingVertical": 10,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "rgb(231, 231, 231)",
          "borderTopLeftRadius": 4,
          "borderTopRightRadius": 4,
        },
        Object {
          "backgroundColor": "transparent",
          "borderColor": "#FFFFFF",
        },
      ]
    }
  >
    <View
      collapsable={false}
      style={
        Object {
          "backgroundColor": "#FFFFFF",
          "bottom": 0,
          "height": 2,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "transform": Array [
            Object {
              "scaleY": 0.5,
            },
          ],
          "zIndex": 1,
        }
      }
    />
    <View
      style={
        Array [
          Object {
            "paddingBottom": 0,
            "paddingTop": 0,
          },
          Object {
            "minHeight": 64,
          },
        ]
      }
    >
      <View
        collapsable={false}
        pointerEvents="none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "opacity": 1,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "transform": Array [
              Object {
                "translateX": 0,
              },
            ],
            "zIndex": 3,
          }
        }
      >
        <Text
          collapsable={false}
          maxFontSizeMultiplier={1.5}
          numberOfLines={1}
          onLayout={[Function]}
          style={
            Object {
              "color": "#FFFFFF",
              "fontFamily": "System",
              "fontSize": 16,
              "fontWeight": undefined,
              "left": 0,
              "opacity": 0,
              "paddingLeft": 12,
              "paddingRight": 12,
              "position": "absolute",
              "textAlign": "left",
              "top": 34,
              "transform": Array [
                Object {
                  "translateX": 0,
                },
                Object {
                  "translateY": 0,
                },
                Object {
                  "scale": 1,
                },
              ],
              "writingDirection": "ltr",
            }
          }
        >
          Label
        </Text>
        <Text
          collapsable={false}
          maxFontSizeMultiplier={1.5}
          numberOfLines={1}
          style={
            Object {
              "color": "#FFFFFF",
              "fontFamily": "System",
              "fontSize": 16,
              "fontWeight": undefined,
              "left": 0,
              "opacity": 0,
              "paddingLeft": 12,
              "paddingRight": 12,
              "position": "absolute",
              "textAlign": "left",
              "top": 34,
              "transform": Array [
                Object {
                  "translateX": 0,
                },
                Object {
                  "translateY": 0,
                },
                Object {
                  "scale": 1,
                },
              ],
              "writingDirection": "ltr",
            }
          }
        >
          Label
        </Text>
      </View>
      <TextInput
        activeOutlineColor="#FFFFFF"
        editable={true}
        maxFontSizeMultiplier={1.5}
        multiline={false}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        placeholder=""
        placeholderTextColor="#FFFFFF"
        selectionColor="#FFFFFF"
        style={
          Array [
            Object {
              "flexGrow": 1,
              "margin": 0,
            },
            Object {
              "paddingLeft": 12,
              "paddingRight": 12,
            },
            Object {
              "height": 64,
            },
            Object {
              "paddingBottom": 4,
              "paddingTop": 24,
            },
            Object {
              "color": "#FFFFFF",
              "fontFamily": "System",
              "fontSize": 16,
              "fontWeight": undefined,
              "textAlign": "left",
              "textAlignVertical": "center",
            },
            false,
            Array [
              Object {},
            ],
          ]
        }
        testID="text-input-flat"
        underlineColorAndroid="transparent"
      />
    </View>
  </View>
</View>
`;

exports[`Test StyledInput Expect StyledInput to be rendered with error text 1`] = `
<View
  style={
    Object {
      "backgroundColor": "transparent",
      "paddingHorizontal": 20,
      "paddingVertical": 10,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "rgb(231, 231, 231)",
          "borderTopLeftRadius": 4,
          "borderTopRightRadius": 4,
        },
        Object {
          "backgroundColor": "transparent",
          "borderColor": "#FF3B30",
        },
      ]
    }
  >
    <View
      collapsable={false}
      style={
        Object {
          "backgroundColor": "#FF3B30",
          "bottom": 0,
          "height": 2,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "transform": Array [
            Object {
              "scaleY": 0.5,
            },
          ],
          "zIndex": 1,
        }
      }
    />
    <View
      style={
        Array [
          Object {
            "paddingBottom": 0,
            "paddingTop": 0,
          },
          Object {
            "minHeight": 64,
          },
        ]
      }
    >
      <TextInput
        activeOutlineColor="#FFFFFF"
        editable={true}
        maxFontSizeMultiplier={1.5}
        multiline={false}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        placeholderTextColor="#FFFFFF"
        selectionColor="#FFFFFF"
        style={
          Array [
            Object {
              "flexGrow": 1,
              "margin": 0,
            },
            Object {
              "paddingLeft": 12,
              "paddingRight": 12,
            },
            Object {
              "height": 64,
            },
            Object {
              "paddingBottom": 0,
              "paddingTop": 0,
            },
            Object {
              "color": "#FF3B30",
              "fontFamily": "System",
              "fontSize": 16,
              "fontWeight": undefined,
              "textAlign": "left",
              "textAlignVertical": "center",
            },
            false,
            Array [
              Object {},
            ],
          ]
        }
        testID="text-input-flat"
        underlineColorAndroid="transparent"
      />
    </View>
  </View>
  <Text
    collapsable={false}
    maxFontSizeMultiplier={1.5}
    onLayout={[Function]}
    style={
      Object {
        "color": "#B00020",
        "fontFamily": "System",
        "fontSize": 12,
        "fontWeight": "400",
        "opacity": 1,
        "paddingHorizontal": 12,
        "paddingVertical": 4,
        "textAlign": "left",
        "transform": Array [
          Object {
            "translateY": -0,
          },
        ],
        "writingDirection": "ltr",
      }
    }
  >
    Some error
  </Text>
</View>
`;

exports[`Test StyledInput Expect StyledInput to not be rendered with error text 1`] = `
<View
  style={
    Object {
      "backgroundColor": "transparent",
      "paddingHorizontal": 20,
      "paddingVertical": 10,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "rgb(231, 231, 231)",
          "borderTopLeftRadius": 4,
          "borderTopRightRadius": 4,
        },
        Object {
          "backgroundColor": "transparent",
          "borderColor": "#FFFFFF",
        },
      ]
    }
  >
    <View
      collapsable={false}
      style={
        Object {
          "backgroundColor": "#FFFFFF",
          "bottom": 0,
          "height": 2,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "transform": Array [
            Object {
              "scaleY": 0.5,
            },
          ],
          "zIndex": 1,
        }
      }
    />
    <View
      style={
        Array [
          Object {
            "paddingBottom": 0,
            "paddingTop": 0,
          },
          Object {
            "minHeight": 64,
          },
        ]
      }
    >
      <TextInput
        activeOutlineColor="#FFFFFF"
        editable={true}
        maxFontSizeMultiplier={1.5}
        multiline={false}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        placeholderTextColor="#FFFFFF"
        selectionColor="#FFFFFF"
        style={
          Array [
            Object {
              "flexGrow": 1,
              "margin": 0,
            },
            Object {
              "paddingLeft": 12,
              "paddingRight": 12,
            },
            Object {
              "height": 64,
            },
            Object {
              "paddingBottom": 0,
              "paddingTop": 0,
            },
            Object {
              "color": "#FFFFFF",
              "fontFamily": "System",
              "fontSize": 16,
              "fontWeight": undefined,
              "textAlign": "left",
              "textAlignVertical": "center",
            },
            false,
            Array [
              Object {},
            ],
          ]
        }
        testID="text-input-flat"
        underlineColorAndroid="transparent"
      />
    </View>
  </View>
  <Text
    collapsable={false}
    maxFontSizeMultiplier={1.5}
    onLayout={[Function]}
    style={
      Object {
        "color": "#B00020",
        "fontFamily": "System",
        "fontSize": 12,
        "fontWeight": "400",
        "opacity": 0,
        "paddingHorizontal": 12,
        "paddingVertical": 4,
        "textAlign": "left",
        "transform": Array [],
        "writingDirection": "ltr",
      }
    }
  >
    Some error
  </Text>
</View>
`;
